Zend/zend_alloc.c

```diff
@@ -329,16 +347,25 @@ struct _zend_mm_heap {
 	double             avg_chunks_count;		/* average number of chunks allocated per request */
 	int                last_chunks_delete_boundary; /* number of chunks after last deletion */
 	int                last_chunks_delete_count;    /* number of deletion over the last boundary */
+#if ZEND_MM_CUSTOM
+	HashTable *tracked_allocs;
+#endif
+	pid_t pid;
+	zend_random_bytes_insecure_state rand_state;
+};
+
+/* This contains security-sensitive data, and is thus mapped as read-only at run-time right after the _zend_mm_heap struct
+ * and accessed via the ZEND_MM_RO_HEAP macro.*/
+struct _zend_mm_ro_heap {
```

Comment:
We could also move the `storage` here, as it's a bit like `custom_heap`: It points to a struct containing function pointers, and it is rarely used.